### ISSUE TITLE
[Inline notification] notification kind icon lacks hover over text customization

### ISSUE THREAD

**Issue by kinueng** [carbon-design-system/carbon#4408]
## Environment

> Operating system

MacOS 10.14.6 (18G103)

> Browser

Chrome Version 77.0.3865.120 (Official Build) (64-bit)

> Automated testing tool and ruleset

Manual testing

> Assistive technology used to verify

Manual testing

## Detailed description

> What version of the Carbon Design System are you using?

    "carbon-components": "9.x"
    "carbon-components-react": "6.x"

> What did you expect to happen?

When setting the hover over text on the close button like so `<InlineNotification kind='error'  iconDescription='Close' ...>`, I was not expecting the error icon to also get `<title>` of `Close`.

> What happened instead?

`<InlineNotification>` should have the ability to to specify text for the error/success/warning icon

> What WCAG 2.1 checkpoint does the issue violate?

Not sure.

## Steps to reproduce the issue

1.  Use mouse pointer to hover over the error/warning/success icon.  See three pictures below for example.  **Note: Warning completely lacks a title**

http://v6-react.carbondesignsystem.com/?selectedKind=Notifications&selectedStory=inline&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs

![image](https://user-images.githubusercontent.com/31117513/67238196-5db14c80-f412-11e9-87c9-8cb84c75f9c1.png)
![image](https://user-images.githubusercontent.com/31117513/67238217-6a35a500-f412-11e9-9e52-c3ac4a1430b5.png)

**Note: This problem is also on the latest carbon react**
http://react.carbondesignsystem.com/?path=/story/notifications--inline

**Comment by asudoh** [comment]
Hi 👋 if you go to KNOBS tab at the bottom or at the right and change the content of  `iconDescription`, you'll see the text is changed. Hope this helps!

**Comment by kinueng** [comment]
Hello @asudoh,
The root problem is the close button (`X` icon) and the Error/Info/Warning icon are sharing the content of `iconDescription`.  This causes the Error/Info/Warning icons to have the wrong `aria-label` and `alt` text.  

Here is a screenshot of an example from here
http://v6-react.carbondesignsystem.com/?selectedKind=Notifications&selectedStory=inline&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel
![image](https://user-images.githubusercontent.com/31117513/67297015-ea114d00-f4ae-11e9-8112-01423e53a3fc.png)

**Comment by kinueng** [comment]
I am advocating for this issue to be reopened with https://github.com/carbon-design-system/carbon/issues/4408#issuecomment-544992407

**Comment by asudoh** [comment]
Thank you for explaining the problem. A PR is welcome to introduce another prop for assistive text for the status icon.

**Comment by joshblack** [comment]
cc @dakahn labelling this as v9 and moving out of v10 milestone
